_posts/en/newsletters/2024-05-08-newsletter.md

```diff
@@ -0,0 +1,315 @@
+---
+title: 'Bitcoin Optech Newsletter #301'
+permalink: /en/newsletters/2024/05/08/
+name: 2024-05-08-newsletter
+slug: 2024-05-08-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter summarizes FIXME
+
+## News
+
+- **Consensus-enforced lamport signatures on top of ECDSA signatures:**
+  Ethan Heilman [posted][] to the Bitcoin-Dev mailing list a method for
+  requiring a transaction be signed by a [lamport signature][] in order
+  to be valid.  This can make spends of P2SH and P2WSH outputs [quantum
+  resistant][] and, [according][poelstra lamport] to Andrew Poelstra
+  means "that size limits are now the only reason that Bitcoin doesn't
+  have covenants."  We'll summarize the protocol below, but to keep our
+  description simple and clear, we will omit several security warnings,
+  so please do not implement anything based on this summary.
+
+  Lamport public keys consist of a two lists of hash digests.  Lamport
+  signatures consist of the preimages for selected hashes.  A program
+  shared between the signer and the validator interprets which preimages
+  are revealed as instructions.  For example, Bob wants to verify that
+  Alice signed a number between 0 and 31 (in binary, 00000 to 11111).
+  Alice private creates a lamport private key from two lists of random
+  numbers:
+
+  ```text
+  private_zeroes = [random(), random(), random(), random(), random()]
+  priavte_ones   = [random(), random(), random(), random(), random()]
+  ```
+
+  She hashes each of those private numbers to create her lamport public
+  key:
+
+  ```text
+  public_zeroes = [private_zeroes[0], ..., private_zeroes[4]]
+  public_ones   = [private_ones[0], ..., private_ones[4]]
```

Comment:
```suggestion
  public_zeroes = [hash(private_zeroes[0]), ..., hash(private_zeroes[4])]
  public_ones   = [hash(private_ones[0]), ..., hash(private_ones[4])]
```

Reply:
Oof, thanks so much for this!  I guess this is why I mainly write documentation, not code!  :-)